src/graph.h

```diff
@@ -91,10 +91,14 @@ class ElementRef {
 // and can be passed as bound inputs to Simit programs.
 class Set {
 public:
-  Set(const std::string &name)
-      : name(name), numElements(0), endpoints(nullptr),
+  enum Kind {Unstructured, LatticeLink};
+  
+  Set(const std::string &name, Kind kind=Unstructured)
```

Comment:
I don't think this constructor needs to take a kind(?) Either it is used to create a non-edge Set or another constructor is used. If we need this constructor it should probably be private.